clang-tools-extra/test/clang-tidy/check_clang_tidy.py

```diff
@@ -172,12 +173,11 @@ def get_prefixes(self):
                 )
 
             if not has_check_fix and not has_check_message and not has_check_note:
-                sys.exit(
-                    "%s, %s or %s not found in the input"
-                    % (self.fixes.prefix, self.messages.prefix, self.notes.prefix)
-                )
+                self.expect_no_diagnosis = True
 
-        assert self.has_check_fixes or self.has_check_messages or self.has_check_notes
+        assert self.expect_no_diagnosis != (
+            self.has_check_fixes or self.has_check_messages or self.has_check_notes
+        )
```

Comment:
What do you think about emitting a message instead of asserting? I.e., something using `sys.ext` like in the other cases that the script exits?